src/libpmemset/pmemset.c

```diff
@@ -297,24 +324,26 @@ pmemset_part_map(struct pmemset_part **part, struct pmemset_extras *extra,
 
 	pmemset_set_persisting_fn(set, part_map);
 
-	/*
-	 * XXX: add multiple part support
-	 */
-	ret = pmemset_insert_part_map(set, part_map);
-	if (ret)
-		goto err_delete_pmap;
-
-	set->previous_part = part_map->desc;
+	if (part_map) {
+		ret = pmemset_insert_part_map(set, part_map);
+		if (ret)
+			goto err_delete_pmap;
+		set->previous_pmap = part_map;
+	}
 
 	if (desc)
-		*desc = part_map->desc;
+		*desc = part_map ? part_map->desc :
```

Comment:
Is anywhere size in desc updated? If no then we have old value of size after map with coalescing.